lib/index.js

```diff
@@ -38,72 +37,64 @@ function setupApp (config, appConfig) {
+        const config = newConfig
         if (err) {
-          Winston.error('Failed to register this mediator, check your config', err)
+          Winston.error('Failed to fetch initial config', err)
           process.exit(1)
-        }
-        apiConf.api.urn = mediatorConfig.urn
-        medUtils.fetchConfig(apiConf.api, (err, newConfig) => {
-          Winston.info('Received initial config:', JSON.stringify(newConfig))
-          const config = newConfig
-          if (err) {
-            Winston.error('Failed to fetch initial config', err)
-            process.exit(1)
-          } else {
-            Winston.info('Successfully registered mediator!')
-            const app = setupApp(config, appConfig)
-            server = app.listen(port, () => {
-              if (apiConf.heartbeat) {
-                const configEmitter = medUtils.activateHeartbeat(apiConf.api)
-                configEmitter.on('error', (err) => {
-                  Winston.error('Heartbeat failed', err)
-                })
-                configEmitter.on('config', (newConfig) => {
-                  Winston.info('Received updated config:', JSON.stringify(newConfig))
+        } else {
+          Winston.info('Successfully registered mediator!')
+          const app = setupApp(config, appConfig)
+          server = app.listen(port, () => {
+            if (apiConf.heartbeat) {
+              const configEmitter = medUtils.activateHeartbeat(apiConf.api)
+              configEmitter.on('error', (err) => {
+                Winston.error('Heartbeat failed', err)
+              })
+              configEmitter.on('config', (newConfig) => {
+                Winston.info('Received updated config:', JSON.stringify(newConfig))
 
-                  // set new config for mediator
-                  Utils.updateConfig(app, config.routes, newConfig.routes)
-                })
-              }
-              callback(null, server)
-            })
-          }
-        })
+                // set new config for mediator
+                Utils.updateConfig(app, config.routes, newConfig.routes)
+              })
+            }
+            callback(null, server)
+          })
+        }
       })
-    } else {
-      // default config
-      const app = setupApp(mediatorConfig.config, appConfig)
-      server = app.listen(port, () => callback(null, server))
-    }
-  })
+    })
+  } else {
+    // default config
+    const app = setupApp(mediatorConfig.config, appConfig)
+    server = app.listen(port, () => callback(null, server))
+  }
 }
 exports.start = start
 
-function setupConfig (callback) {
-  Confit(Path.join(__dirname, '..', 'config')).create(callback)
-}
-exports.setupConfig = setupConfig
-
 function stop (callback) {
   server.close(callback)
 }
 exports.stop = stop
 
 if (!module.parent) {
+  const normalConfig = config.configure()
+  if (normalConfig && normalConfig.openhim && normalConfig.openhim.register === 'false') {
```

Comment:
@Zooloo2014 This step would be better handled by converting the config value to boolean immediately in the config rather that passing the string around

Reply:
Good suggestion.